rust/signed_doc/src/signature/kid/mod.rs

```diff
@@ -0,0 +1,114 @@
+//! COSE Signature Protected Header `kid`.
+mod authority;
+mod key_version;
+mod role;
+mod role0_pk;
+
+use std::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+use authority::Authority;
+use key_version::KeyVersion;
+use role::Role;
+use role0_pk::Role0PublicKey;
+
+/// Catalyst Signed Document Key ID
+///
+/// Key ID associated with a `COSE` Signature that is structured as a Universal Resource
+/// Identifier (`URI`).
+#[derive(Debug, Clone)]
+pub struct Kid {
```

Comment:
Probably should rename this type to `KidUri` and the whole URI type also probably belongs in `catalyst-types` as it is general, and could be used in multiple places.